src/ray/core_worker/core_worker.cc

```diff
@@ -459,15 +459,44 @@ CoreWorker::CoreWorker(const CoreWorkerOptions &options, const WorkerID &worker_
       {{"worker_id", worker_id.Hex()}});
 }
 
+CoreWorker::~CoreWorker() { RAY_LOG(INFO) << "Core worker is destructed"; }
+
 void CoreWorker::Shutdown() {
-  io_service_.stop();
+  RAY_LOG(INFO) << "Shutting down a core worker.";
+  is_shutdown_ = true;
```

Comment:
Would it be useful to have `RAY_CHECK(is_shutdown_.exchange(true));` to make sure `Shutdown()` is called only once, or return early if `is_shutdown_` is already `true`? Ok to leave a TODO if it makes sense.

Reply:
I think it is safer to guarantee this is called only once. But I am not confident it won't break with the current code.

Just make the method idempotent for now!

Reply:
I also added a log message